### PR TITLE
Problems with TFT5

### PR DESCRIPTION
**Pull request by Guillem96**
As far as we are concerned, Tensorflow T5 has some problems:

- Shifting inputs right should add at the beginning of the sequences a *Start of Sequence token* (padding token in T5 case), but currently, it generates a useless sequence of zeros. This is due to this line of code:

```python
shifted_input_ids = tf.zeros_like(input_ids, dtype=tf.int32) # No matter what, this generates zeros
# ... carry this 0s all over the function
```

*Note*

We observe, that PyTorch's `T5ForConditionalGeneration` loss is a scalar. Contrary, with the `TFT5ForConditionalGeneration`, the loss is a tensor where each loss element corresponds to a *non-ignored* label. We think that the loss should be reduced. Is there any reason for keeping the loss *as is* instead of reducing it?

### CODE REVIEW

File: src/transformers/modeling_tf_t5.py [hunk 1]
```diff
@@ -1069,7 +1067,7 @@ def call(
 
 
 @add_start_docstrings("""T5 Model with a `language modeling` head on top. """, T5_START_DOCSTRING)
-class TFT5ForConditionalGeneration(TFT5PreTrainedModel, TFCausalLanguageModelingLoss):
+class TFT5ForConditionalGeneration(TFT5PreTrainedModel, TFCausalLanguageModeling ):
```
Comment (LysandreJik) [inline src/transformers/modeling_tf_t5.py:1070]: I think this should stay as `TFCausalLanguageModelingLoss `

File: test-hf.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import transformers
+import tensorflow as tf
+
+model = transformers.TFT5ForConditionalGeneration.from_pretrained('t5-small')
+input_ids = tf.random.uniform([1, 512], maxval=1000, dtype=tf.int32)
+labels = tf.random.uniform([1, 512], maxval=1000, dtype=tf.int32)
+
+loss, *_ = model(dict(input_ids=input_ids, labels=labels))
+print(loss)
```
Comment (LysandreJik) [inline test-hf.py:9]: What is this test?

**Review by LysandreJik — CHANGES_REQUESTED** [review]

File: src/transformers/modeling_tf_t5.py (hunk 1 above)
Comment (Guillem96) [reply src/transformers/modeling_tf_t5.py:1070]: Sorry I forgot to delete this quick test

File: test-hf.py (hunk 2 above)
Comment (Guillem96) [reply test-hf.py]: Sorry I forgot to remove it

File: src/transformers/modeling_tf_t5.py (hunk 1 above)
Comment (Guillem96) [reply src/transformers/modeling_tf_t5.py]: This is true I unintentionally change it

**Comment by patrickvonplaten** [comment]
Closing in favor of #6214
